pkg/e2e/infra/utils.go

```diff
@@ -0,0 +1,65 @@
+package infra
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	e2e "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/framework"
+	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	controllernode "github.com/openshift/cluster-api/pkg/controller/node"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func isOneMachinePerNode(client runtimeclient.Client) bool {
+	listOptions := runtimeclient.ListOptions{
+		Namespace: e2e.TestContext.MachineApiNamespace,
+	}
+	machineList := mapiv1beta1.MachineList{}
+	nodeList := corev1.NodeList{}
+
+	if err := wait.PollImmediate(5*time.Second, e2e.WaitShort, func() (bool, error) {
+		if err := client.List(context.TODO(), &listOptions, &machineList); err != nil {
+			glog.Errorf("Error querying api for machineList object: %v, retrying...", err)
+			return false, nil
+		}
+		if err := client.List(context.TODO(), &listOptions, &nodeList); err != nil {
+			glog.Errorf("Error querying api for nodeList object: %v, retrying...", err)
+			return false, nil
+		}
+
+		glog.Infof("Expecting the same number of machines and nodes, have %d nodes and %d machines", len(nodeList.Items), len(machineList.Items))
+		if len(machineList.Items) != len(nodeList.Items) {
+			return false, nil
+		}
+
+		nodeNameToMachineAnnotation := make(map[string]string)
```

Comment:
Why not create a function that does this? Can we / will we do this elsewhere?

Reply:
I'll do later if it happens to be reused